crates/cxx-qt-lib-extras/src/core/qlocale.rs

```diff
@@ -0,0 +1,570 @@
+
+        /// Returns the language of this locale.
+        // QLocale::Language QLocale::language() const
+        #[rust_name = "language"]
+        fn language(self: &QLocale) -> QLocaleLanguage;
+
+        /// Returns the two- or three-letter language code for language, as defined in the ISO 639 standards.
+        // QString QLocale::languageToCode(QLocale::Language language, QLocale::LanguageCodeTypes codeTypes = AnyLanguageCode)
+
+        /// Returns a QString containing the name of language.
+        // static QString QLocale::languageToString(QLocale::Language language)
+        // #[rust_name = "language_to_string"]
+        // fn languageToString(language: QLocaleLanguage) -> QString;
+
+        // QList<QLocale> QLocale::matchingLocales(QLocale::Language language, QLocale::Script script, QLocale::Territory territory)
+
+        /// Returns the measurement system for the locale.
+        // QLocale::MeasurementSystem QLocale::measurementSystem() const
+        #[rust_name = "measurement_system"]
+        fn measurementSystem(self: &QLocale) -> QLocaleMeasurementSystem;
+
+        /// Returns the localized name of month, in the format specified by type.
+        // QString QLocale::monthName(int month, QLocale::FormatType type = LongFormat) const
+        #[rust_name = "month_name"]
+        fn monthName(self: &QLocale, month: i32, formatType: QLocaleFormatType) -> QString;
+
+        /// The short name of this locale.
+        // QString QLocale::name(QLocale::TagSeparator separator = TagSeparator::Underscore) const
+        #[rust_name = "name"]
+        fn name(self: &QLocale, separator: QLocaleTagSeparator) -> QString;
+
+        /// Returns a native name of the language for the locale. For example "Schweizer Hochdeutsch" for the Swiss-German locale.
+        #[rust_name = "native_language_name"]
+        fn nativeLanguageName(self: &QLocale) -> QString;
+
+        /// Returns a native name of the territory for the locale. For example "España" for Spanish/Spain locale.
+        #[rust_name = "native_territory_name"]
+        fn nativeTerritoryName(self: &QLocale) -> QString;
+
+    }
+
+    #[namespace = "rust::cxxqtlib1"]
+    unsafe extern "C++" {
+
+        type QLocaleTagSeparator;
+        type QLocaleCurrencySymbolFormat;
+        type QLocaleFormatType;
+        type QLocaleLanguage;
+        type QLocaleMeasurementSystem;
+
+        include!("cxx-qt-lib/common.h");
+
+        #[doc(hidden)]
+        #[rust_name = "qlocale_init_default"]
+        fn construct() -> QLocale;
+
+        #[doc(hidden)]
+        #[allow(dead_code)]
+        #[rust_name = "qlocale_init_from_name"]
+        fn construct(name: &QString) -> QLocale;
+    }
+}
+
+pub use ffi::QLocaleCurrencySymbolFormat;
+pub use ffi::QLocaleFormatType;
+pub use ffi::QLocaleLanguage;
+pub use ffi::QLocaleMeasurementSystem;
+pub use ffi::QLocaleTagSeparator;
+
+#[repr(C)]
+pub struct QLocale {
+    _cspec: MaybeUninit<usize>,
+}
+
+impl Default for QLocale {
```

Comment:
As QLocale is not trivially copyable or destructible can you add a `impl Clone for QLocale` and a `impl Drop for QLocale` implementation similar to other places (note we have templates for those too eg see how QString is done)